go.mod

```diff
@@ -7,7 +7,7 @@ require (
 	github.com/adamthesax/grpc-proxy v0.0.0-20220525203857-13e92d14f87a
 	github.com/armon/go-metrics v0.4.1
 	github.com/hashicorp/consul-server-connection-manager v0.1.3
-	github.com/hashicorp/consul/proto-public v0.4.0
+	github.com/hashicorp/consul/proto-public v0.1.2-0.20230817221841-aac2dac759c6
```

Comment:
Is this intentional?

Reply:
I think the answer is yes because she is pulling in an unreleased version of this module with all the new V2 APIs. We do something similar in `consul-k8s` with new versions of the `api` module until it is released. Go does a weird thing where it uses the last tag that exists on the `main` branch along with part of the SHA to identify the commit to pull in.